lib/discourse_ip_info.rb

```diff
@@ -16,27 +22,45 @@ def initialize
   end
 
   def lookup(ip)
-    return {} unless @mmdb
+    ret = {}
 
-    begin
-      result = @mmdb.lookup(ip)
-    rescue
-      Rails.logger.error("IP #{ip} could not be looked up in MaxMindDB.")
+    if @loc_mmdb
+      begin
+        result = @loc_mmdb.lookup(ip)
+        if result&.found?
+          ret[:country] = result.country.name
+          ret[:country_code] = result.country.iso_code
+          ret[:region] = result.subdivisions.most_specific.name
+          ret[:city] = result.city.name
+        end
+      rescue
+        Rails.logger.error("IP #{ip} could not be looked up in MaxMind GeoLite2-City database.")
+      end
     end
 
-    return {} if !result || !result.found?
+    if @asn_mmdb
+      begin
+        result = @asn_mmdb.lookup(ip)
+        if result&.found?
+          result = result.to_hash
+          ret[:asn] = result["autonomous_system_number"]
+          ret[:organization] = result["autonomous_system_organization"]
+        end
+      rescue
+        Rails.logger.error("IP #{ip} could not be looked up in MaxMind GeoLite2-ASN database.")
+      end
+    end
 
-    {
-      country: result.country.name,
-      country_code: result.country.iso_code,
-      region: result.subdivisions.most_specific.name,
-      city: result.city.name,
-    }
+    begin
+      result = Resolv::DNS.new.getname(ip)
```

Comment:
Is this code ran in test, in general this is one of the super rare times I would be ok with mocking